vllm/lora/layers.py

```diff
@@ -508,14 +514,30 @@ def slice_lora_a(self, lora_a: torch.Tensor) -> torch.Tensor:
         return lora_a
 
     def slice_lora_b(self, lora_b: torch.Tensor) -> torch.Tensor:
-        tensor_model_parallel_rank = get_tensor_model_parallel_rank()
-        shard_size = self.output_dim
-        start_idx = tensor_model_parallel_rank * shard_size
-        end_idx = (tensor_model_parallel_rank + 1) * shard_size
-        lora_b = lora_b[:, start_idx:end_idx]
+        # Applicable to cases where the base_layer is
+        # MergedColumnParallelLinear.
+        if self.is_merged_col_linear:
+            tp_rank = get_tensor_model_parallel_rank()
+            shard_size = self.output_size // 2
+            offset = lora_b.shape[-1] // 2
+
+            left_weight = lora_b[:, tp_rank * shard_size:(tp_rank + 1) *
+                                 shard_size]
+            right_weigt = lora_b[:, offset + tp_rank * shard_size:offset +
+                                 (tp_rank + 1) * shard_size]
+            lora_b = torch.cat([left_weight, right_weigt], dim=1)
+        # Applicable to cases where the base_layer is
+        # ColumnParallelLinear.
+        else:
+            tensor_model_parallel_rank = get_tensor_model_parallel_rank()
+            shard_size = self.output_dim
+            start_idx = tensor_model_parallel_rank * shard_size
+            end_idx = (tensor_model_parallel_rank + 1) * shard_size
+            lora_b = lora_b[:, start_idx:end_idx]
         return lora_b
 
     def slice_bias(self, bias: torch.Tensor) -> torch.Tensor:
+        # TODO: Fix the slicing logic of bias.
```

Comment:
Maybe it's because you haven't implemented this yet?

Reply:
No, I don't plan to address the bias slicing logic in this PR. I have doubts about the bias implementation, as I haven't fully understood it yet.

Reply:
For this test [script](https://github.com/vllm-project/vllm/blob/main/tests/lora/test_llama.py), even with 4 GPUs, the test still fails. The main issue is that the test keeps hanging."

Reply:
If it's about the test hanging, maybe some of the workers failed to call `torch.distributed.all_gather`.